torchxrayvision/datasets.py

```diff
@@ -195,7 +197,24 @@ def __len__(self):
     def __getitem__(self, idx):
         return self.dataset[self.idxs[idx]]
 
-    
+
+class TarDataset(Dataset):
+    def __init__(self, imgpath):
+        if imgpath.endswith(".tar"):
+            self.tarred = tarfile.open(imgpath)
+            self.tar_paths = self.tarred.getmembers()
```

Comment:
So you said this takes a lot of time. I see how this approach is super robust. I did some tests for time and it seems like just 30 seconds for the MIMIC data. What about caching this using a dict based on the file path? It would speed things up if multiple objects are created? But it seems like a reasonable price to pay.
The alternative is to skip it and assume a file structure but that is not as nice.

Reply:
Ok, using a dict, the second load is around 10x faster on my machine. 

The dict could also be pickled so there is only one slow load. That option would lead to issues if someone wanted to change the tarfile, although I don't know why they would do that.

Reply:
Doing that hashing could be nice but also could be annoying to debug and create more issues (like if there are no write permissions).